(window.webpackJsonp=window.webpackJsonp||[]).push([[90],{440:function(t,e,o){"use strict";o.r(e);var s=o(42),a=Object(s.a)({},(function(){var t=this,e=t.$createElement,o=t._self._c||e;return o("ContentSlotsDistributor",{attrs:{"slot-key":t.$parent.slotKey}},[o("h1",{attrs:{id:"my-tasks-suddently-stopped"}},[o("a",{staticClass:"header-anchor",attrs:{href:"#my-tasks-suddently-stopped"}},[t._v("#")]),t._v(" My tasks suddently stopped")]),t._v(" "),o("p",[t._v("For you to know that your tasks stopped in the middle of it, you most likely have the "),o("code",[t._v("LOG_LEVEL")]),t._v(" higher than 20, or the actual tasks is not giving the expected output.")]),t._v(" "),o("h2",{attrs:{id:"situations-that-it-might-have-stopped"}},[o("a",{staticClass:"header-anchor",attrs:{href:"#situations-that-it-might-have-stopped"}},[t._v("#")]),t._v(" Situations that it might have stopped")]),t._v(" "),o("p",[t._v("Rated from high to low probability")]),t._v(" "),o("ol",[o("li",[o("em",[o("strong",[t._v("Issue:")])]),t._v(" Most of the time the tasks didn't stop, it's just chilling waiting for your rate limits to update a bucket. "),o("em",[o("strong",[t._v("Solution:")])]),t._v(" Wait it out or ask a higher limit in the Riot Dev Portal.")]),t._v(" "),o("li",[o("em",[o("strong",[t._v("Issue:")])]),t._v(" You caused memory leakage hitting the OS per process RAM limit and python aborted some of the objects. "),o("em",[o("strong",[t._v("Solution:")])]),t._v(" Avoid loading all the objects into the memory (python runtime), create a function or method that can consume the object while loading it and garbage it afterwards. There is a detailed warning and examples in both "),o("strong",[o("RouterLink",{attrs:{to:"/core/gatherer.html"}},[t._v("Gatherer")])],1),t._v(" and "),o("strong",[o("RouterLink",{attrs:{to:"/core/queue.html"}},[t._v("Queue")])],1),t._v(".")]),t._v(" "),o("li",[o("em",[o("strong",[t._v("Issue:")])]),t._v(" There is a synchronous task running that blocks the event loop giving no chance at all to the task. "),o("em",[o("strong",[t._v("Solution:")])]),t._v(" Nuke the task, and make it run in another process or thread.")]),t._v(" "),o("li",[o("em",[o("strong",[t._v("Issue:")])]),t._v(" It is very unlikely that Pyot itself will cause your tasks to stop, but it's not warrantied to be a non-zero chance from happening. "),o("em",[o("strong",[t._v("Solution:")])]),t._v(" Try cleaning the redis db if you're using "),o("code",[t._v("RedisLimiter")]),t._v(" for your rate limits and make a clean up of the storages that pyot interacts with.")]),t._v(" "),o("li",[o("em",[o("strong",[t._v("Issue:")])]),t._v(" You're running the thing in a potato that suddenly went out of power or crashed. "),o("em",[o("strong",[t._v("Solution:")])]),t._v(" Run it in an actual computer.")])])])}),[],!1,null,null,null);e.default=a.exports}}]);